nucypher/utilities/networking.py

```diff
@@ -15,42 +15,167 @@
     """
-    Attempts to automatically get the external IP from ifconfig.me
-    If the request fails, it falls back to the standard process.
+    Utility function to send a GET request to a URL returning it's
+    text content or None, suppressing all errors. Certificate is
+    needed if the remote URL source is self-signed.
     """
     try:
-        rest_host = get_external_ip_from_centralized_source()
-    except UnknownIPAddress:
-        if force:
-            raise
+        # 'None' or 'True' will verify self-signed certificates
+        response = requests.get(url, verify=certificate)
+    except RequestErrors:
+        return None
+    if response.status_code == 200:
+        return response.text
+
+
+def get_external_ip_from_default_teacher(network: str,
+                                         federated_only: bool = False,
+                                         log: Logger = IP_DETECTION_LOGGER,
+                                         registry: BaseContractRegistry = None
+                                         ) -> Union[str, None]:
+    from nucypher.characters.lawful import Ursula
+
+    if federated_only and registry:
+        raise ValueError('Federated mode must not be true if registry is provided.')
+    base_error = 'Cannot determine IP using default teacher'
+    try:
+        top_teacher_url = RestMiddleware.TEACHER_NODES[network][0]
+    except IndexError:
+        log.debug(f'{base_error}: No teacher available for network "{network}".')
+        return
+    except KeyError:
+        log.debug(f'{base_error}: Unknown network "{network}".')
+        return
+
+    ####
+    # TODO: Clean this mess #1481
+    node_storage = LocalFileBasedNodeStorage(federated_only=federated_only)
+    Ursula.set_cert_storage_function(node_storage.store_node_certificate)
+    Ursula.set_federated_mode(federated_only)
+    #####
+
+    teacher = Ursula.from_teacher_uri(teacher_uri=top_teacher_url,
+                                      federated_only=federated_only,
+                                      min_stake=0)  # TODO: Handle customized min stake here.
+
+    # TODO: Pass registry here to verify stake (not essential here since it's a hardcoded node)
+    client = NucypherMiddlewareClient()
+    try:
+        response = client.get(node_or_sprout=teacher, path=f"ping", timeout=2)  # TLS certificate logic within
+    except RestMiddleware.UnexpectedResponse:
+        # 404, 405, 500, All server response codes handled by will be caught here.
+        return  # Default teacher does not support this request - just move on.
+    if response.status_code == 200:
+        try:
+            ip = str(ip_address(response.text))
+        except ValueError:
+            error = f'Default teacher at {top_teacher_url} returned an invalid IP response; Got {response.text}'
+            raise UnknownIPAddress(error)
+        log.info(f'Fetched external IP address from default teacher ({top_teacher_url} reported {ip}).')
```

Comment:
Maintaining the same format as output in `get_external_ip_from_known_nodes` and `get_external_ip_from_centralized_source`.

```suggestion
        log.info(f'Fetched external IP address ({ip}) from default teacher ({top_teacher_url}).')
```

Reply:
:heavy_check_mark: